core/modules/filters/eachday.js

```diff
@@ -24,9 +24,16 @@ exports.eachday = function(source,operator,options) {
 		value = (new Date(value)).setHours(0,0,0,0);
 		return value+0;
 	};
+	// Convert ISO date string to Date object if needed.
+	var toDateObj = function(value) {
+		if(!(value instanceof Date)) {
+			value = new Date($tw.utils.parseDate(value));
+		}
+		return value;
+	};
 	source(function(tiddler,title) {
 		if(tiddler && tiddler.fields[fieldName]) {
-			var value = toDate(tiddler.fields[fieldName]);
+			var value = toDate(toDateObj(tiddler.fields[fieldName]));
```

Comment:
Ah I see my mistake there.  I will update with some new commits hopefully
later in the day.

On Wed, Mar 11, 2015 at 3:26 PM, Jeremy Ruston <notifications@github.com>
wrote:

> In core/modules/filters/eachday.js
> <https://github.com/Jermolene/TiddlyWiki5/pull/1520#discussion_r26257422>:
>
> >  	source(function(tiddler,title) {
> >  		if(tiddler && tiddler.fields[fieldName]) {
> > -			var value = toDate(tiddler.fields[fieldName]);
> > +			var value = toDate(toDateObj(tiddler.fields[fieldName]));
>
> I think we can remove the toDateObj() function and instead change this
> line to:
>
> var value = toDate($tw.utils.parseDate(tiddler.fields[fieldName]));
>
> The reason is that both toDate() and isSameDay() convert their argument
> to a date if necessary, and $tw.utils.parseDate() copes correctly with an
> argument that is already a date object.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jermolene/TiddlyWiki5/pull/1520/files#r26257422>.
>



-- 
Roma Quintin Hicks